extensions/smallrye-jwt/runtime/src/main/java/io/quarkus/smallrye/jwt/runtime/auth/MpJwtValidator.java

```diff
@@ -52,25 +44,16 @@ public MpJwtValidator(JWTAuthContextInfo authContextInfo) {
     public CompletionStage<SecurityIdentity> authenticate(TokenAuthenticationRequest request,
             AuthenticationRequestContext context) {
         try {
-            JwtContext jwtContext = parser.parse(request.getToken().getToken(), authContextInfo);
+            JsonWebToken jwt = parser.parse(request.getToken().getToken());
```

Comment:
Why are these changes necessary ? I'd like to keep an option open to have the factory used instead of the parser so that people can plugin custom factories.

Reply:
I don't see why this should not be possible any more. People can now easily inject the JwtParser implementation they like. And with the changes I proposed in https://github.com/smallrye/smallrye-jwt/pull/194, it would also easily be possible to inject a different `JWTCallerPrincipalFactory` into the `DefaultJwtParser`.